c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -5451,6 +5474,437 @@ bool constructor(JSContext *cx, unsigned argc, Value *vp) {
 }
 } // namespace TextDecoder
 
+#ifdef ENABLE_OBJECT_STORE
+namespace ObjectStoreEntry {
+namespace Slots {
+enum {
+  Body = RequestOrResponse::Slots::Body,
+  BodyStream = RequestOrResponse::Slots::BodyStream,
+  HasBody = RequestOrResponse::Slots::HasBody,
+  BodyUsed = RequestOrResponse::Slots::BodyUsed,
+  Count
+};
+};
+
+const unsigned ctor_length = 0;
+
+bool check_receiver(JSContext *cx, HandleValue receiver, const char *method_name);
+
+template <BodyReadResult result_type> bool bodyAll(JSContext *cx, unsigned argc, Value *vp) {
+  METHOD_HEADER(0)
+  return RequestOrResponse::bodyAll<result_type>(cx, args, self);
+}
+
+bool body_get(JSContext *cx, unsigned argc, Value *vp) {
+  METHOD_HEADER(0)
+  return RequestOrResponse::body_get(cx, args, self, true);
+}
+
+bool bodyUsed_get(JSContext *cx, unsigned argc, Value *vp) {
+  METHOD_HEADER(0)
+  args.rval().setBoolean(RequestOrResponse::body_used(self));
+  return true;
+}
+
+const JSFunctionSpec methods[] = {
+    JS_FN("arrayBuffer", bodyAll<BodyReadResult::ArrayBuffer>, 0, JSPROP_ENUMERATE),
+    JS_FN("json", bodyAll<BodyReadResult::JSON>, 0, JSPROP_ENUMERATE),
+    JS_FN("text", bodyAll<BodyReadResult::Text>, 0, JSPROP_ENUMERATE), JS_FS_END};
+
+const JSPropertySpec properties[] = {JS_PSG("body", body_get, JSPROP_ENUMERATE),
+                                     JS_PSG("bodyUsed", bodyUsed_get, JSPROP_ENUMERATE), JS_PS_END};
+
+bool constructor(JSContext *cx, unsigned argc, Value *vp) {
+  JS_ReportErrorUTF8(cx, "ObjectStoreEntry can't be instantiated directly");
+  return false;
+}
+
+CLASS_BOILERPLATE(ObjectStoreEntry)
```

Comment:
The reason we do not use CLASS_BOILERPLATE_NO_CTOR is because that macro will also remove the constructor from the application environment, which makes `entry instanceof ObjectStoreEntry` not be possible because `ObjectStoreEntry` would be undefined